From 91e4de46b6f665bfdac04ba78de2bb0150293437 Mon Sep 17 00:00:00 2001
From: Hunter Liu <htliu@linkedsemi.com>
Date: Mon, 19 Aug 2024 22:36:35 +0800
Subject: [PATCH 3/3] irq-lingsi-peri:     1. remove gc lock from
 irq_get_irqchip_state     2. remove irq ack for newer FPGA version

---
Upstream-Status: Pending

 drivers/irqchip/irq-lingsi-peri.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/irqchip/irq-lingsi-peri.c b/drivers/irqchip/irq-lingsi-peri.c
index 6d0c6a24cd35..30afbd25e062 100644
--- a/drivers/irqchip/irq-lingsi-peri.c
+++ b/drivers/irqchip/irq-lingsi-peri.c
@@ -205,9 +205,6 @@ static int lingsi_irq_set_irqchip_state(struct irq_data *d,
 
     switch (which) {
     case IRQCHIP_STATE_PENDING:
-        if (!val) {
-            iowrite32(BIT(hwirq), base + lingsi_bank->clr);
-        }
         break;
 
     case IRQCHIP_STATE_MASKED:
@@ -231,8 +228,6 @@ static int lingsi_irq_get_irqchip_state(struct irq_data *d,
     uint32_t hwirq = irqd_to_hwirq(d) % IRQS_PER_BANK;
     unsigned long data;
 
-    irq_gc_lock(gc);
-
     switch (which) {
     case IRQCHIP_STATE_PENDING:
         data = ioread32(base + lingsi_bank->raw);
@@ -248,7 +243,6 @@ static int lingsi_irq_get_irqchip_state(struct irq_data *d,
         return -EINVAL;
     }
 
-    irq_gc_unlock(gc);
     return 0;
 }
 
@@ -291,11 +285,11 @@ static int __init lingsi_peri_init(const struct lingsi_peri_drv_data *drv_data,
         gc->chip_types->type = IRQ_TYPE_EDGE_BOTH;
         gc->chip_types->chip.irq_unmask = irq_gc_mask_set_bit;
         gc->chip_types->chip.irq_mask = irq_gc_mask_clr_bit;
-        gc->chip_types->chip.irq_ack = irq_gc_ack_set_bit;
+        /* 💀 never set .irq_ack to NULL. 💀 */
+        gc->chip_types->chip.irq_ack = irq_gc_noop;
         gc->chip_types->chip.irq_get_irqchip_state = lingsi_irq_get_irqchip_state,
         gc->chip_types->chip.irq_set_irqchip_state = lingsi_irq_set_irqchip_state,
         gc->chip_types->regs.mask = lingsi_bank->mask;
-        gc->chip_types->regs.ack = lingsi_bank->clr;
         gc->private = (void *)chip_data;
     }
     nr_irqs = of_irq_count(node);
-- 
2.39.2

